packages/openlayers/lib/map/apply-context-diff.ts

```diff
@@ -26,12 +26,13 @@ export function applyContextDiffToMap(
 
   // insert added layers
   for (const layerAdded of contextDiff.layersAdded) {
-    const layer = createLayer(layerAdded.layer);
-    if (layerAdded.position >= layers.getLength()) {
-      layers.push(layer);
-    } else {
-      layers.insertAt(layerAdded.position, layer);
-    }
+    createLayer(layerAdded.layer).then(layer => {
```

Comment:
I think this approach might cause race conditions when creating and adding layers; the original assumption was that layers would be added in the same order as they appeared on the `layersAdded` array. We can either use `async` or change the logic to create all layers in parallel and then adding them in the map.